### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by Muzammilshabbir**
## Summary by CodeRabbit

- **New Features**
  - Added detailed role-based permissions and descriptions for team members.
  - Introduced comprehensive team member management UI with search, filter, and edit capabilities.
  - Implemented new API endpoints for team members, clinicians, clinical info, licenses, clinician services, billing documents, statements, superbills, permissions, and analytics with filtering, pagination, and CSV export.
  - Enabled secure file uploads and management for client groups with metadata tracking.
  - Added a reusable sidebar component for editing team member details.
  - Enhanced clinician and service management with create/update flows and validations.
  - Added extensive React components for editing and displaying team member personal, clinical, license, and service information.
  - Introduced integration and unit tests for all new API endpoints and UI components.
  - Added comprehensive service module with React Query hooks for team member and clinician data management.

- **Bug Fixes**
  - Improved session-based authorization enforcement and error handling across APIs.
  - Enhanced test suite isolation with expanded database cleanup.
  - Fixed data retrieval logic to better handle nested data and reduce flakiness in tests.

- **Tests**
  - Added extensive unit and integration tests covering all new and existing API endpoints, including edge cases and error scenarios.
  - Updated test guidelines to enforce continuous test runs until all pass.

- **Database**
  - Added new tables for permissions, role-permission mappings, client group files, statements, superbills, and related indexes and constraints.
  - Extended existing models with richer relations and metadata fields to support new features.

- **Documentation**
  - Expanded testing guidelines with explicit instructions for request creation and API imports.

- **Chores**
  - Modularized seed scripts for users and permissions with automated permission assignment to roles.
  - Refactored helper functions to include additional clinician details for improved data access.

### CODE REVIEW

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

File: .cursor/rules/rules-for-writing-tests.mdc [hunk 1]
```diff
@@ -31,6 +31,8 @@ Unit tests focus on testing individual functions or methods in isolation, with a
 - **Coverage**: Test all code paths, including error cases.
 - **Example**: `apps/back-office/__tests__/api/location/route.unit.test.ts`
 
+- Keep running `npx vitest apps/back-office/__tests__/api/**test-path**` in a loop **until all test cases pass without any failure**. Exit only when the test result is fully green (i.e., no failing or pending tests).
```
Comment (kaza) [inline .cursor/rules/rules-for-writing-tests.mdc:34]: love this rule

File: apps/back-office/__tests__/api/analytics/appointmentStatus/route.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+          id: uuidv4(),
+          legal_first_name: "Test",
+          legal_last_name: "Client",
+          is_active: true,
+        },
+      });
+
+      // Create client group membership
+      await prisma.clientGroupMembership.create({
+        data: {
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/analytics/appointmentStatus/route.integration.test.ts:65]: use our data mocks that we already created.

File: apps/back-office/__tests__/api/analytics/route.unit.test.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    invoice: {
+      aggregate: vi.fn(),
+    },
+    appointment: {
+      count: vi.fn(),
+    },
+    surveyAnswers: {
+      count: vi.fn(),
+    },
+  },
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/analytics/route.unit.test.ts:30]: use our prismaMock

File: apps/back-office/__tests__/api/analytics/route.unit.test.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+import { prisma } from "@mcw/database";
+import { startOfMonth, endOfMonth, subMonths } from "date-fns";
+import { Prisma } from "@prisma/client";
+import { createRequest } from "@mcw/utils";
+
+// Mock the logger
+vi.mock("@mcw/logger", () => ({
+  logger: {
+    error: vi.fn(),
+  },
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/analytics/route.unit.test.ts:12]: do we really need to mock the logger?

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Review by talhabalaj — CHANGES_REQUESTED** [review]

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

File: apps/back-office/__tests__/api/billing-documents/route.unit.test.ts [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+    (prisma.invoice.findMany as unknown as Mock).mockResolvedValue(
+      mockInvoices,
+    );
+    (prisma.superbill.findMany as unknown as Mock).mockResolvedValue(
+      mockSuperbills,
+    );
+    (prisma.statement.findMany as unknown as Mock).mockResolvedValue(
+      mockStatements,
+    );
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.unit.test.ts:146]: use prismaMock.

File: apps/back-office/__tests__/api/billing-documents/route.unit.test.ts [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+  const createdAt = new Date("2023-01-25T12:00:00Z");
+  return {
+    id: "statement-1",
+    statement_number: 2001,
+    client_group_id: "group-1",
+    created_at: createdAt,
+    beginning_balance: new Decimal(250),
+    invoices_total: new Decimal(100),
+    payments_total: new Decimal(75),
+    ending_balance: new Decimal(275),
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.unit.test.ts:106]: use our data mockers.

File: apps/back-office/__tests__/api/billing-documents/route.unit.test.ts [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+vi.mock("@mcw/database", () => {
+  const invoiceFindManyMock = vi.fn();
+  const superbillFindManyMock = vi.fn();
+  const statementFindManyMock = vi.fn();
+
+  return {
+    prisma: {
+      invoice: {
+        findMany: invoiceFindManyMock,
+      },
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.unit.test.ts:30]: use prismaMock

File: apps/back-office/__tests__/api/permissions/route.integration.test.ts [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+    error: vi.fn(),
+    warn: vi.fn(),
+    debug: vi.fn(),
+  }),
+}));
+
+// Get references to the mocks
+const { getBackOfficeSession } = await import("@/utils/helpers");
+// We import the logger but don't use it directly in the integration tests
+await import("@mcw/logger");
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/permissions/route.integration.test.ts:30]: why do you need this

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.
